src/c++/perf_analyzer/test_profile_data_exporter.cc

```diff
@@ -244,4 +248,114 @@ TEST_CASE("profile_data_exporter: OutputToFile")
   }
 }
 
+TEST_CASE("profile_data_exporter: AddServiceKind")
+{
+  MockProfileDataExporter exporter{};
+  exporter.ClearDocument();
+
+  SUBCASE("Backend kind: TRITON")
+  {
+    cb::BackendKind service_kind = cb::BackendKind::TRITON;
+    exporter.AddServiceKind(service_kind);
+
+    std::string json{R"({ "service_kind": "triton" })"};
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_kind{expected_document["service_kind"]};
+    const rapidjson::Value& actual_kind{exporter.document_["service_kind"]};
+    CHECK(actual_kind == expected_kind);
+  }
+
+  SUBCASE("Backend kind: TENSORFLOW_SERVING")
+  {
+    cb::BackendKind service_kind = cb::BackendKind::TENSORFLOW_SERVING;
+    exporter.AddServiceKind(service_kind);
+
+    std::string json{R"({ "service_kind": "tfserving" })"};
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_kind{expected_document["service_kind"]};
+    const rapidjson::Value& actual_kind{exporter.document_["service_kind"]};
+    CHECK(actual_kind == expected_kind);
+  }
+
+  SUBCASE("Backend kind: TORCHSERVE")
+  {
+    cb::BackendKind service_kind = cb::BackendKind::TORCHSERVE;
+    exporter.AddServiceKind(service_kind);
+
+    std::string json{R"({ "service_kind": "torchserve" })"};
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_kind{expected_document["service_kind"]};
+    const rapidjson::Value& actual_kind{exporter.document_["service_kind"]};
+    CHECK(actual_kind == expected_kind);
+  }
+
+  SUBCASE("Backend kind: TRITON_C_API")
+  {
+    cb::BackendKind service_kind = cb::BackendKind::TRITON_C_API;
+    exporter.AddServiceKind(service_kind);
+
+    std::string json{R"({ "service_kind": "triton_c_api" })"};
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_kind{expected_document["service_kind"]};
+    const rapidjson::Value& actual_kind{exporter.document_["service_kind"]};
+    CHECK(actual_kind == expected_kind);
+  }
+
+  SUBCASE("Backend kind: OPENAI")
+  {
+    cb::BackendKind service_kind = cb::BackendKind::OPENAI;
+    exporter.AddServiceKind(service_kind);
+
+    std::string json{R"({ "service_kind": "openai" })"};
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_kind{expected_document["service_kind"]};
+    const rapidjson::Value& actual_kind{exporter.document_["service_kind"]};
+    CHECK(actual_kind == expected_kind);
+  }
+}
+
+TEST_CASE("profile_data_exporter: AddEndpoint")
+{
+  MockProfileDataExporter exporter{};
+  exporter.ClearDocument();
+
+  SUBCASE("Endpoint: OpenAI Chat Completions")
```

Comment:
Same suggestion for here, try to move the duplicated logic outside of the subcase

Reply:
Updated.